core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceBuilderTest.java

```diff
@@ -191,6 +183,52 @@ public void root(@Header("a") ArrayList<String> a,
                              @Header("a") LinkedList<String> b) {}
         });
 
+        // Multiple paths are allowed
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Post
+            @Path("/a")
+            public void root() {}
+        });
+
+        Server.builder().annotatedService(new Object() {
+            @Post("/")
+            @Get("/")
+            public void root() {}
+        });
+
+        // Multiple paths with matching params
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Path("/a/{param}")
+            @Path("/b/{param}")
+            public void root(@Param("param") String param) {}
+        });
+
+        // Multiple paths with non matching but optional path variables
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Path("/a")
+            @Path("/b/{param}")
+            public void root(@Param("param") Optional<String> param) {}
+        });
+
+        // Multiple paths with non matching non-optional path variables
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Path("/a/{param1}")
+            @Path("/b/{param2}")
+            public void root(String param1, String param2) {}
+        });
+
+        // Multiple paths with same value
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Path("/a")
+            @Path("/a")
```

Comment:
Not sure that this is ok or not. Isn't it better to raise an exception in this case?

Reply:
I'm not sure too :-) But I think throwing an exception is too strict because most people expect the same result `@Path("/a")`?

Reply:
Yes, it's too strict. I have thought about the case when a user wants to use this same annotation and there was none. If he/she does this, I thought he/she wants to use a different path but didn't notice it. So maybe raising an exception would be better. But It's fine as it is though.

Reply:
I understand what you meant. which one is better? 🧐

Reply:
What do you think @jrhee17? If you think this is too much or hard to implement, I think we can just leave this as it is. 😄

Reply:
Behavior-wise, I think warn/throwing an exception both are fine. I really have no pref. for this.

I think we should consider the following though:

1) If we stick with detecting strictly equal `ExactPathMapping`s, I think this can be do-able. If we consider, cases like `@Path("/path/{name}")`, `@Path("/path/{name2}")`, or bring `RegexPathMapping` into the mix, things might become complicated :(
2) Different methods with the same `PathMapping`. For consistent behavior, I feel like the following should also warn/throw. Let me know what you guys think :)

```java
    @Get("/duplicatePath")
    public HttpResponse test1() { return HttpResponse.of(200); }

    @Get("/duplicatePath")
    public HttpResponse test2() { return HttpResponse.of(200); }
```

Reply:
I just realized that we have [rejectedRouteHandler](https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L190) for this. So let's leave as it is. 😄

Reply:
Oooh  I didn't know about this.   Let me fix the current impl. so that `rejectedRouteHandler` can take care of duplicate routes. 🙇